src/utils.ts

```diff
@@ -444,3 +446,38 @@ export function applyPathMapping(
 
     return sourcePath;
 }
+
+/**
+ * Gets the localized resources that will be displayed in the frontend
+ * @param extensionPath The root folder for the extension path.
+ */
+export async function getLocalizedStrings(extensionPath: string): Promise<string> {
+    try {
+        const settings = vscode.workspace.getConfiguration(SETTINGS_STORE_NAME);
+        const localizationEnabled: string = settings.get("Localization") || SETTINGS_LOCALIZATION;
+        if (localizationEnabled === "Enabled" || localizationEnabled === "qps-ploc") {
+            let locale = "en-us";
+            if (process.env.VSCODE_NLS_CONFIG) {
+                locale = StringsProvider.getFallback(JSON.parse(process.env.VSCODE_NLS_CONFIG).locale);
+            }
+
+            // override for pseudo
+            if (localizationEnabled === "qps-ploc") {
+                locale = "qps-ploc";
+            }
+
+            if (locale !== "en-us") {
+                const resourcePath = vscode.Uri.file(
+                    path.join(extensionPath, "resources", "locales", `${locale}.json`));
+                const frontendStringDocument = await vscode.workspace.openTextDocument(resourcePath);
+                if (frontendStringDocument) {
+                    return frontendStringDocument.getText();
+                }
+            }
+        }
+
+        return "";
+    } catch (e) {

```

Comment:
Can we scope down the try/catch to only the part that is actually loading the file?